core/tracker/tracker.go

```diff
@@ -125,29 +125,73 @@ func (t *Tracker) Run(ctx context.Context) error {
-// If it failed, it also returns the component where it failed and a human friendly error message explaining why.
-func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
-	events := make([]event, len(es))
-	copy(events, es)
+// analyseDutyFailed detects if the given duty failed. It returns false if the duty didn't fail, i.e., the duty didn't get stuck and completes the sigAgg component.
+// It returns true if the duty failed. It also returns the component where the duty got stuck and a human friendly error message explaining why.
+func analyseDutyFailed(duty core.Duty, es map[core.Duty][]event) (bool, component, string) {
+	// dutyFailedComponent returns true if the duty failed. It also returns the component where the duty got stuck.  If the duty didn't get stuck, it
+	// returns the sigAgg component. It assumes that all the input events are for a single duty.
+	dutyFailedComponent := func(es []event) (bool, component) {
+		events := make([]event, len(es))
+		copy(events, es)
+
+		// Sort in reverse order (see order above).
+		sort.Slice(events, func(i, j int) bool {
+			return events[i].component > events[j].component
+		})
+
+		if len(events) == 0 {
+			return false, sentinel
+		}
+
+		c := events[0].component
+		if c == sigAgg {
+			// Duty completed successfully
+			return false, sigAgg
+		}
+
+		return true, c
+	}
 
-	// Sort in reverse order (see order above).
-	sort.Slice(events, func(i, j int) bool {
-		return events[i].component > events[j].component
-	})
+	var (
+		failed bool
+		comp   component
+		msg    string
+	)
 
-	if len(events) == 0 {
-		return false, sentinel, "No events to analyse"
+	failed, comp = dutyFailedComponent(es[duty])
+	if !failed {
+		return false, comp, msg
 	}
 
-	if events[0].component == sigAgg {
-		// Duty completed successfully
-		return false, sigAgg, ""
+	switch comp {
+	case scheduler:
+		msg = "couldn't schedule duty. Either there is a problem resolving the duty or there are no active validators"
+	case fetcher:
+		if duty.Type == core.DutyProposer || duty.Type == core.DutyBuilderProposer {
+			// Proposer duties may fail if core.DutyRandao fails
+			f, _ := dutyFailedComponent(es[core.NewRandaoDuty(duty.Slot)])
+			if f {
+				// core.DutyRandao failed
+				msg = "couldn't propose duty as randao failed due to insufficient partial signatures"
+			}
+		} else {
+			msg = "couldn't fetch duty as connection to BN maybe down"
+		}
+	case consensus:
+		msg = "couldn't reach quorum. Either there are insufficient peers or some peers are byzantine"
+	case validatorAPI:
+		msg = "either VC didn't send partial signature or connection to BN is down"
+	case parSigDBInternal:
+		msg = "couldn't save partially signed duty data or there was a problem calling parsigex"
+	case parSigEx:
+		msg = "couldn't exchange partial signatures. Either there aren't enough peers or some peers sent invalid sigs"
+	case parSigDBThreshold:
+		msg = "couldn't reach threshold of partial signatures"
```

Comment:
`insufficient partial signatures received, minimum required threshold not reached`